bin/reth/src/cli/config.rs

```diff
@@ -16,7 +20,47 @@ pub trait RethRpcConfig {
     /// The configured ethereum RPC settings.
     fn eth_config(&self) -> EthConfig;
 
-    // TODO extract more functions from RpcServerArgs
+    /// Returns the max request size in bytes.
+    fn rpc_max_request_size_bytes(&self) -> u32;
+
+    /// Returns the max response size in bytes.
+    fn rpc_max_response_size_bytes(&self) -> u32;
+
+    /// Extracts the gas price oracle config from the args.
+    fn gas_price_oracle_config(&self) -> GasPriceOracleConfig;
+
+    /// Creates the [TransportRpcModuleConfig] from cli args.
+    ///
+    /// This sets all the api modules, and configures additional settings like gas price oracle
+    /// settings in the [TransportRpcModuleConfig].
+    fn transport_rpc_module_config(&self) -> TransportRpcModuleConfig;
+
+    /// Returns the default server builder for http/ws
+    fn http_ws_server_builder(&self) -> ServerBuilder;
+
+    /// Returns the default ipc server builder
+    fn ipc_server_builder(&self) -> IpcServerBuilder;
+
+    /// Creates the [RpcServerConfig] from cli args.
+    fn rpc_server_config(&self) -> RpcServerConfig;
+
+    /// Creates the [AuthServerConfig] from cli args.
+    fn auth_server_config(&self, jwt_secret: JwtSecret) -> Result<AuthServerConfig, RpcError>;
+
+    /// The execution layer and consensus layer clients SHOULD accept a configuration parameter:
+    /// jwt-secret, which designates a file containing the hex-encoded 256 bit secret key to be used
+    /// for verifying/generating JWT tokens.
+    ///
+    /// If such a parameter is given, but the file cannot be read, or does not contain a hex-encoded
+    /// key of 256 bits, the client SHOULD treat this as an error.
+    ///
+    /// If such a parameter is not given, the client SHOULD generate such a token, valid for the
+    /// duration of the execution, and SHOULD store the hex-encoded secret as a jwt.hex file on
+    /// the filesystem. This file can then be used to provision the counterpart client.
+    ///
+    /// The `default_jwt_path` provided as an argument will be used as the default location for the
+    /// jwt secret in case the `auth_jwtsecret` argument is not provided.
+    fn jwt_secret(&self, default_jwt_path: PathBuf) -> Result<JwtSecret, JwtError>;
```

Comment:
I'm not quite happy with how we have currently implemented this, because rn this always requires a path, even if there might be one configured via cli args, so maybe we can find a better solution for this eventually, but for now this is fine.

Reply:
@mattsse Just opened #4419 accordingly.